src/bootstrap/src/utils/cmd.rs

```diff
@@ -0,0 +1,92 @@
+use std::ffi::OsStr;
+use std::path::Path;
+use std::process::{Command, ExitStatus, Output};
+
+/// Wrapper around `std::process::Command` whose execution will be eventually
+/// tracked.
+#[derive(Debug)]
+pub struct TrackedCommand {
+    cmd: Command,
+}
+
+impl TrackedCommand {
+    pub fn new<P: AsRef<OsStr>>(program: P) -> Self {
+        Self { cmd: Command::new(program) }
+    }
+
+    pub fn current_dir<P: AsRef<Path>>(mut self, path: P) -> Self {
+        self.cmd.current_dir(path);
+        self
+    }
+
+    pub fn arg<A: AsRef<OsStr>>(mut self, arg: A) -> Self {
+        self.cmd.arg(arg);
+        self
+    }
+
+    /// Runs the command, ensuring that it has succeeded.
+    #[track_caller]
+    pub fn run(&mut self) -> CommandOutput {
+        let output = self.run_maybe();
+        if !output.is_success() {
+            panic!(
+                "`{:?}`\nwas supposed to succeed, but it failed with {}\nStdout: {}\nStderr: {}",
+                self.cmd,
+                output.status,
+                output.stdout(),
+                output.stderr()
+            )
+        }
+        output
+    }
+
+    /// Runs the command.
+    #[track_caller]
+    pub fn run_maybe(&mut self) -> CommandOutput {
+        let output = self.cmd.output().expect("Cannot execute process");
```

Comment:
Suggestion: it might be nice to report the complete command invocation that failed